lib/Myriad/Service/Storage/Remote.pm

```diff
@@ -0,0 +1,73 @@
+package Myriad::Service::Storage::Remote;
+
+use Myriad::Class;
+
+# VERSION
+# AUTHORITY
+
+=encoding utf8
+
+=head1 NAME
+
+Myriad::Service::Storage::Remote - abstraction to access other services storage.
+
+=head1 SYNOPSIS
+
+ my $storage = $api->service_by_name('service')->storage;
+ await $storage->get('some_key');
+
+=head1 DESCRIPTION
+
+=cut
+
+has $prefix;
+has $storage;
+
+BUILD (%args) {
+    $prefix = delete $args{prefix} // die 'need a prefix';
+    $storage = delete $args{storage} // die 'need a storage instance';
+}
+
+=head2 apply_prefix
+
+Maps the requested key into the service's keyspace
+so we can pass it over to the generic storage layer.
+
+Takes the following parameters:
+
+=over 4
+
+=item * C<$k> - the key
+
+=back
+
+Returns the modified key.
+
+=cut
+
+method apply_prefix ($k) {
+    return $prefix . '.' . $k;
+}
+
+async method get ($k, %args) { return await $storage->get($self->apply_prefix($k)) }
```

Comment:
at this point I think there's too much duplication, let's move this method list out to somewhere else